src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -143,60 +160,123 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
+                    .append("], total_space [")
+                    .append(new ByteSizeValue(nodePath.fileStore.getTotalSpace()))
+                    .append("], spins? [")
+                    .append(spinsDesc)
+                    .append("], mount [")
+                    .append(nodePath.fileStore)
+                    .append("], type [")
+                    .append(nodePath.fileStore.type())
+                    .append(']');
             }
-            logger.info(sb.toString());
+            logger.debug(sb.toString());
+        } else if (logger.isInfoEnabled()) {
+            FsStats.Info totFSInfo = new FsStats.Info();
+            Set<String> allTypes = new HashSet<>();
+            Set<String> allSpins = new HashSet<>();
+            Set<String> allMounts = new HashSet<>();
+            for (NodePath nodePath : nodePaths) {
+                // TODO: can/should I use the chosen FsProbe instead (i.e. sigar if it's available)?
+                FsStats.Info fsInfo = JmxFsProbe.getFSInfo(nodePath);
+                String mount = fsInfo.getMount();
+                if (allMounts.contains(mount) == false) {
+                    allMounts.add(mount);
+                    String type = fsInfo.getType();
+                    if (type != null) {
+                        allTypes.add(type);
+                    }
+                    Boolean spins = fsInfo.getSpins();
+                    if (spins == null) {
+                        allSpins.add("unknown");
+                    } else if (spins.booleanValue()) {
+                        allSpins.add("possibly");
+                    } else {
+                        allSpins.add("no");
+                    }
+                    totFSInfo.add(fsInfo);
+                }
+            }
+
+            // Just log a 1-line summary:
+            logger.info(String.format(Locale.ROOT,
+                                      "using [%d] data paths, net usable_space [%s], net total_space [%s], spins? [%s], types [%s]",
+                                      nodePaths.length,
+                                      totFSInfo.getAvailable(),
+                                      totFSInfo.getTotal(),
+                                      toString(allSpins),
+                                      toString(allTypes)));
         }
+    }
 
-        this.nodeIndicesPaths = new Path[nodePaths.length];
-        for (int i = 0; i < nodePaths.length; i++) {
-            nodeIndicesPaths[i] = nodePaths[i].resolve(INDICES_FOLDER);
+    private static String toString(Collection<String> items) {
+        StringBuilder b = new StringBuilder();
+        for(String item : items) {
+            if (b.length() > 0) {
+                b.append(", ");
+            }
+            b.append(item);
         }
+        return b.toString();
     }
 
+
+    // TODO: move somewhere more "util"?  But, this is somewhat hacky code ... not great to publicize it any more:
```

Comment:
can the methods like getFileStore() be package private? I don't see anything outside of this file using them.